.github/workflows/guard-main-origin.yml

```
@@ -0,0 +1,26 @@
+    branches: [main]            # PRs whose *target* is main only
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  ensure-valid-origin:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR is from allowed branch
+        run: |
```

actionlint (1.7.10)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)